server/uploader.js

```diff
@@ -0,0 +1,132 @@
+/**
+ * @fileoverview upload function handler
+ */
+
+'use strict'
+
+var path;
+var fs = require('fs');
+var multer = require('multer');
+var shell = require('shelljs');
+
+var utils = require('./utils');
+var binding = require('./binding');
+var segtree = require('./segtree');
+
+var upload = multer();
+
+module.exports = {
+
+  /**
+   * Uploads a file or a directory to server.
+   * @param req {Object} Including query parameders.
```

Comment:
Looks like this order is still incorrect? should be 
@param {Object} req Including query parameters, and ditto for all below.